### PR TITLE
Glossary Entity Added - Ingestion fix

### LINKED ISSUE #2791: Ingestion fails due to CreateGlossaryRequest
**Affected module**
Ingestion Framework

**Describe the bug**
Getting following error while ingestion.
```
Traceback (most recent call last):
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/bin/metadata", line 8, in <module>
    sys.exit(metadata())
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/metadata/cmd.py", line 72, in ingest
    workflow = Workflow.create(workflow_config)
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/metadata/ingestion/api/workflow.py", line 150, in create
    return cls(config)
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/metadata/ingestion/api/workflow.py", line 68, in __init__
    self.source: Source = source_class.create(
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/metadata/ingestion/source/sample_usage.py", line 52, in create
    return cls(config, metadata_config, ctx)
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/metadata/ingestion/source/sample_usage.py", line 44, in __init__
    self.service = get_database_service_or_create(
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/metadata/utils/helpers.py", line 60, in get_database_service_or_create
    service = metadata.get_by_name(entity=DatabaseService, fqdn=config.service_name)
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/metadata/ingestion/ometa/ometa_api.py", line 376, in get_by_name
    return self._get(entity=entity, path=f"name/{fqdn}", fields=fields)
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/metadata/ingestion/ometa/ometa_api.py", line 401, in _get
    resp = self.client.get(f"{self.get_suffix(entity)}/{path}{fields_str}")
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/metadata/ingestion/ometa/ometa_api.py", line 174, in get_suffix
    entity, get_args(Union[Glossary, self.get_create_entity_type(Glossary)])
  File "/Users/sachinchaurasiya/Desktop/OpenMetadata/ingestion/env/lib/python3.9/site-packages/metadata/ingestion/ometa/ometa_api.py", line 319, in get_create_entity_type
    create_class = getattr(
AttributeError: module 'metadata.generated.schema.api.data.createGlossary' has no attribute 'CreateGlossaryRequest'
```

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes :

fix #2791
Ingestion fails due to CreateGlossaryRequest not being in the ometa_api.

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@harshach @pmbrull

### CODE REVIEW

File: ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+from metadata.generated.schema.entity.data.glossary import Glossary
+from metadata.ingestion.ometa.client import APIError
+
+T = TypeVar("T", bound=BaseModel)  # pylint: disable=invalid-name
+logger = logging.getLogger(__name__)
+
+
+class GlossaryMixin:
+    def list_glossaries(
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py:19]: We already have a `list_entities` function. Doesn't it work for `Glossary`?

File: ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        except APIError as err:
+            logger.error(f"GET {entity.__name__}. Error {err.status_code} - {err}")
+            return None
+
+    def create_glossaries_category(self, glossaries_body: Glossary):
+        """Method to create new Glossary category
+        Args:
+            glossaries_body (Glossary): body of the request
+        """
+        path = "/glossaries"
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py:42]: We should take this from `get_suffix`

**Review by pmbrull — COMMENTED** [review]
Thanks for the PR! A couple of comments pls

**Comment by harshach** [comment]
@ayush-shah we don't need glossary as another index. I am removing the code https://github.com/open-metadata/OpenMetadata/pull/2818
we can turn this PR to retain the API to list and post glossary

File: ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py (hunk 1 above)
Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py]: I have removed the method as discussed and can be done using `list_entities`

File: ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py (hunk 2 above)
Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py]: Made that Change, Thanks

**Review by harshach — APPROVED** [review]

**Pull request merged**